tests/adapters/switches/juniper_test.py

```diff
@@ -4297,6 +4297,68 @@ def test_enable_interface_succeeds(self):
 
         self.switch.set_interface_state("ge-0/0/6", ON)
 
+    def test_reset_interface_succeeds_removing_enabled_state(self):
+
+        self.netconf_mock.should_receive("edit_config").once().with_args(target="candidate", config=is_xml("""
+            <config>
+              <configuration>
+                <interfaces>
+                  <interface>
+                    <name>ge-0/0/6</name>
+                    <enable />
+                  </interface>
+                </interfaces>
+              </configuration>
+            </config>
+        """)).and_return(an_ok_response())
+
+        self.switch.set_interface_state("ge-0/0/6", ON)
+
+        self.netconf_mock.should_receive("edit_config").once().with_args(target="candidate", config=is_xml("""
+            <config>
+              <configuration>
+                <interfaces>
+                  <interface>
+                    <name>ge-0/0/6</name>
+                  </interface>
+                </interfaces>
+              </configuration>
+            </config>
+        """)).and_return(an_ok_response())
+
+        self.switch.reset_interface_state("ge-0/0/6")
```

Comment:
this is NOT how juniper works, at all

first of all : there's no need to do these 2 commands in a unit test if they are mocked.  there's no state to the mock

second : to remove a member from the tree you need to give it operation="delete"

See unset_interface_native_vlan for reference